HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/DotNetZipArchiveProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Compression;
+using HyperVExtension.Extensions;
+using Windows.Storage;
+
+namespace HyperVExtension.Models.VirtualMachineCreation;
+
+/// <summary>
+/// Provides methods to extract a zip archive. This class uses the .NET <see cref="ZipArchive"/> class to extract the contents of the archive file.
+/// <see cref="ZipArchive"/> is used by Hyper-V Manager's VM Quick Create feature. This gives us parity, but in the future it is expected that faster/more efficient
+/// archive extraction libraries will be added.
+/// </summary>
+public sealed class DotNetZipArchiveProvider : IArchiveProvider
+{
+    // Same buffer size used by Hyper-V Manager's VM gallery feature.
+    private readonly int _transferBufferSize = 4096;
```

Comment:
Does `4096` come from documentation?

Reply:
Its the exact size that the quick create window in Windows is using. We're trying to be in parity with that feature. There is no public documentation explaining why they chose this amount. However, I added the comment as a FYI